bundles/mapping/mapmodule/MapModuleClass.ol.js

```diff
@@ -530,32 +531,173 @@ export class MapModule extends AbstractMapModule {
+                duration: duration
+            }, callback);
+            break;
+        case 'fly':
+            const flyZoom = view.getZoom();
+            view.animate({
+                center: location,
+                duration: duration
+            }, callback);
+            view.animate({
+                zoom: flyZoom - 1,
+                duration: duration / 2
+            }, {
+                zoom: flyZoom,
+                duration: duration / 2
+            }, callback);
+            break;
+        case 'zoomPan':
+            view.animate({
+                center: location,
+                zoom: zoom,
+                duration: duration,
+                easing: easeOut
+            }, callback);
+            break;
+        default:
+            view.setCenter(location);
+            view.setZoom(zoom);
+            break;
+        }
+        return true;
+    }
+
     /**
      * @method centerMap
      * Moves the map to the given position and zoomlevel.
      * @param {Number[] | Object} lonlat coordinates to move the map to
-     * @param {Number} zoomLevel absolute zoomlevel to set the map to
+     * @param {Object | Number} zoomLevel absolute zoomlevel to set the map to
      * @param {Boolean} suppressEnd true to NOT send an event about the map move
      *  (other components wont know that the map has moved, only use when chaining moves and
      *     wanting to notify at end of the chain for performance reasons or similar) (optional)
+     * @param {Object} options options, such as animation and duration
+     *     Usable animations: fly/pan/zoomPan
      * @return {Boolean} success
      */
-    centerMap (lonlat, zoom, suppressEnd) {
+    centerMap (lonlat, zoom, suppressEnd, options) {
+        const view = this.getMap().getView();
+        const animation = options && options.animation ? options.animation : '';
+        const duration = options && options.duration ? options.duration : 3000;
+        // If no animation, notify without duration
+        const notifyDuration = animation === '' ? 0 : duration;
+
         lonlat = this.normalizeLonLat(lonlat);
         if (!this.isValidLonLat(lonlat.lon, lonlat.lat)) {
             return false;
         }
-        this.getMap().getView().setCenter([lonlat.lon, lonlat.lat]);
+
         if (zoom === null || zoom === undefined) {
-            zoom = this.getMapZoom();
+            zoom = { type: 'zoom', value: this.getMapZoom() };
         }
-        this.getMap().getView().setZoom(zoom);
+        if (zoom === Number) {
+            zoom = { type: 'zoom', value: zoom };
+        }
+        const zoomValue = zoom.type === 'scale' ? view.getZoomForResolution(zoom.value) : zoom.value;
+
+        this._animateTo(lonlat, zoomValue, animation, duration);
+
         this.updateDomain();
         if (suppressEnd !== true) {
-            this.notifyMoveEnd();
+            setTimeout(() => {
+                this.notifyMoveEnd();
+            }, notifyDuration);
         }
```

Comment:
Just realized that we need to call updateDomain() _after_ the map has moved. So do something like this:
```suggestion
    setTimeout(() => {
        this.updateDomain();
        if (suppressEnd !== true) {
            this.notifyMoveEnd();
        }  
   }, notifyDuration);
```
Otherwise the domain object will have the location of the map _before_ the animation starts and not what the location is _after_ the animation.

Reply:
Scratch that. This https://github.com/oskariorg/oskari-frontend/blob/1.53.1/bundles/mapping/mapmodule/MapModuleClass.ol.js#L101 gets triggered with the animate method so the events and domain get updated.